iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/NetworkImpl.java

```diff
@@ -995,8 +994,8 @@ private void replaceDanglingLineByLine(List<MergedLine> lines, Multimap<Boundary
                 la.setNode2(mergedLine.node2);
             }
             TieLineImpl l = la.add();
-            l.setCurrentLimits(Side.ONE, (CurrentLimitsImpl) mergedLine.limits1);
-            l.setCurrentLimits(Side.TWO, (CurrentLimitsImpl) mergedLine.limits2);
+            l.getLimitsHolder1().setOperationalLimits(LimitType.CURRENT, mergedLine.limits1);
+            l.getLimitsHolder2().setOperationalLimits(LimitType.CURRENT, mergedLine.limits2);
```

Comment:
I would hide the getLimitsHolder from the API, because it's not really user friendly